### PR TITLE
Add optional filtering on `version` in reviews list API

### PR DESCRIPTION
**Pull request by diox**
Needed to work around #5006

### CODE REVIEW

File: src/olympia/api/paginator.py [hunk 1]
```diff
@@ -89,3 +89,23 @@ def get_paginated_response(self, data):
...
     django_paginator_class = ESPaginator
+
+
+class OneOrZeroPageNumberPagination(CustomPageNumberPagination):
+    """Fake pagination that returns a result object like
+    CustomPageNumberPagination, but for special cases where we know we're never
+    going to have more than one object in the results anyway.
+    """
+    def paginate_queryset(self, queryset, request, view=None):
+        return list(queryset[:1])
```
Comment (EnTeQuAk) [inline src/olympia/api/paginator.py]: I honestly wonder if we should have an `assert` here that makes sure the queryset doesn't contain more than one value? I feel like failing silently is quite bad if the used view is only supposed to return one value.

File: src/olympia/reviews/tests/test_views.py [hunk 2]
```diff
@@ -996,6 +996,137 @@ def test_list_addon_and_user(self):
...
+        review_deleted = Review.objects.create(
+            addon=self.addon, body='review deleted', user=self.user)
+        review_deleted.delete()
+        Review.objects.create(
+            addon=other_review_same_addon.addon, body='reply to other user',
+            reply_to=other_review_same_addon, user=self.user)
+        Review.objects.create(
+            addon=addon_factory(), body='review from other addon',
+            user=self.user)
+
```
Comment (EnTeQuAk) [inline src/olympia/reviews/tests/test_views.py]: Damn we really need factories to simplify this process and make it more readable 😨

File: src/olympia/reviews/tests/test_views.py [hunk 3]
```diff
@@ -1182,6 +1313,54 @@ def test_list_weird_parameters(self):
...
+            self.url, {'addon': self.addon.pk, 'version': u'çæ→'})
+        assert response.status_code == 400
+        data = json.loads(response.content)
+        assert data == {'detail': 'version parameter should be an integer.'}
+
+    # settings_test sets CACHE_COUNT_TIMEOUT to -1 and it's too late to
+    # override it, so instead mock the TIMEOUT property in cache-machine.
+    @mock.patch('caching.config.TIMEOUT', 300)
+    def test_get_then_post_then_get_any_caching_is_cleared(self):
+        """Make sure there is no overzealous caching going when requesting the
```
Comment (EnTeQuAk) [inline src/olympia/reviews/tests/test_views.py]: "caching going on"

File: src/olympia/reviews/tests/test_views.py [hunk 4]
```diff
@@ -1182,6 +1313,54 @@ def test_list_weird_parameters(self):
...
+        assert response.status_code == 200
+        data = json.loads(response.content)
+        assert len(data['results']) == 0
+        assert data['count'] == 0
+
+        # Do a post to add a review by this user.
+        response = self.client.post(self.url, {
+            'addon': self.addon.pk, 'body': u'test bodyé', 'title': u'blahé',
+            'rating': 5, 'version': self.addon.current_version.pk},
+            REMOTE_ADDR='213.225.312.5')
```
Comment (EnTeQuAk) [inline src/olympia/reviews/tests/test_views.py]: Do you need to supply the `REMOTE_ADDR`?

File: src/olympia/api/paginator.py (hunk 1 above)
Comment (diox) [inline src/olympia/api/paginator.py]: I don't know about a hard fail like an `assert`. I could add some logging maybe? Not sure it's worth it. I could envision this pagination class be used in a cases where you do have more than 1 object but you don't care.

File: src/olympia/reviews/tests/test_views.py (hunk 4 above)
Comment (diox) [inline src/olympia/reviews/tests/test_views.py]: Nope, I'll remove it

**Pull request merged**